cmd/prometheus/main.go

```diff
@@ -938,8 +948,17 @@ func main() {
 		)
 	}
 	{
+		exporterLogger := log.With(logger, "component", "gcm_exporter")
```

Comment:
question: Is there any way to wrap this up in a constructor, i.e. `newExporter` call?

It looks like the client (i.e. Prometheus) has a bit more explicit setup work to do now than before, which is less than ideal in a world where we're trying to defork ourselves. Is it feasible to consolidate these in a constructor in our export package?

A lot of these functions look new to me and I'm not exactly sure how they work, so I may just be out of the loop.

Reply:
Done. Thanks!